syntax = "proto3";

import "stream_data.proto";

message InitRequest{
    string runtime = 1;
    string args_json = 2;
}

message Index{
    oneof index{
        int32 index_i = 1;
        string index_s = 2;
    }
}

message SetInputRequest{
    oneof data{
        Index index = 1;
        Chunk np_arr_chunk = 2;
    }
}

message BenchmarkRequest{
    int32 warmup = 1;
    int32 repeat = 2;
    int32 number = 3;
}

message BenchmarkResponse{
    float mean_ts = 1;
    float std_ts = 2;
    float max_ts = 3;
    float min_ts = 4;
}

message GetOutputRequest{
    int32 index = 1;
}

message GetOutputResponse{
    bytes np_data = 1;
}

message Empty {};

message DetailsResponse{
    string json = 1;
}
